mpc/share-conversion-core/src/fields/gf2_128.rs

```diff
@@ -0,0 +1,195 @@
+//! This module implements the extension field GF(2^128)
+
+use super::Field;
+use mpc_core::Block;
+use rand::{distributions::Standard, prelude::Distribution};
+use std::ops::{Add, Mul, Neg};
+
+#[derive(Copy, Clone, Debug, PartialOrd, Ord, PartialEq, Eq)]
+pub struct Gf2_128(pub(crate) u128);
+
+impl Gf2_128 {
+    pub fn new(input: u128) -> Self {
+        Gf2_128(input)
+    }
+
+    pub fn into_inner(self) -> u128 {
+        self.0
+    }
+}
+
+impl From<Gf2_128> for Block {
+    fn from(value: Gf2_128) -> Self {
+        Block::new(value.0)
+    }
+}
+
+impl From<Block> for Gf2_128 {
+    fn from(value: Block) -> Self {
+        Gf2_128::new(value.inner())
+    }
+}
+
+impl Distribution<Gf2_128> for Standard {
+    fn sample<R: rand::Rng + ?Sized>(&self, rng: &mut R) -> Gf2_128 {
+        Gf2_128(self.sample(rng))
+    }
+}
+
+impl Add for Gf2_128 {
+    type Output = Self;
+
+    #[allow(clippy::suspicious_arithmetic_impl)]
+    fn add(self, rhs: Self) -> Self::Output {
+        Self(self.0 ^ rhs.0)
+    }
+}
+
+impl Mul for Gf2_128 {
+    type Output = Self;
+
+    /// Galois field multiplication of two 128-bit blocks reduced by the GCM polynomial
+    fn mul(self, rhs: Self) -> Self::Output {
+        /// R is the GCM polynomial in big-endian. In hex: "000000000000000000000000000000E1"
```

Comment:
/// R is the GCM "special element" (see section 2.5 of "The Galois/Counter Mode of Operation (GCM)")
/// in big-endian. In hex: "000000000000000000000000000000E1"